subt/cloudsim2osgar.py

```diff
@@ -190,10 +192,17 @@ def __init__(self, robot_name, robot_config, robot_is_marsupial):
         # main thread receives data from osgar and sends it to ROS
         while True:
             channel, data = self.bus.listen()
-            rospy.logdebug("receiving: {} {}".format(channel, data))
             # switch on channel to feed different ROS publishers
             if channel in publishers:
-                publishers[channel].publish(*data)  # just guessing for Empty, where ([],) is wrong
+                if channel == 'cmd_vel':
```

Comment:
A proposal that would help us avoid another if-elif-elif-elif-elif-... growing over our heads, especially given that there already is a dict/map that we use when handling incoming messages:
Make `publishers` a map from channel name to `(handler, publisher)` pair and instead of this if-elif, have:
```
publisher, handler = publishers[channel]
publisher.publish(handler(data))
```

Reply:
OK, this makes sense - I will create fixup (in 1hour, I hope).

Reply:
I deliberately offered a way which allows a single `def empty(): return Empty()` shared handler used across different channels. Unless publisher.publish(Empty()) *does not work*, in which case I would step back for now.

I would really like this code not to end up like rosmsg not so long ago or main.py still. The best time to prevent that is now :-)

Reply:
Yes, as far as I remember `publisher.publish(Empty())` does not work - I will re-try it and if it works then Yes, let's do it. Thanks for support :).